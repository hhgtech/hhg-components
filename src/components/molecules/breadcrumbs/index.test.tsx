import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Breadcrumbs } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const dropdown = render(
    <MockTheme>
      <Breadcrumbs
        size="md"
        pages={[
          {
            label: 'Home',
            value: '/',
          },
          {
            label: 'Category 1',
            value: '/category-1',
          },
          {
            label:
              'Sub Category 1 with a very long name that will not show completely',
            value: '/category-1/sub-category-1',
          },
          {
            label: 'Sub Category 2',
            value: '/category-1/sub-category-1/sub-category-2',
          },
        ]}
      />
    </MockTheme>,
  )

  const self = dropdown.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toContainHTML('Home')
  expect(self).toContainHTML('/category-1')
})
