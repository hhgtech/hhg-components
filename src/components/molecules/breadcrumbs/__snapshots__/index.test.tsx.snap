// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > .collapsed-breadcrumb {
  border: none;
}

.c0 > .collapsed-breadcrumb > div {
  padding: 0;
}

.c0 > .collapsed-breadcrumb > ul {
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
}

.c0[data-size='lg'] {
  font-size: 16px;
  line-height: 24px;
  font-weight: 700;
}

.c0[data-size='md'] {
  font-size: 14px;
  line-height: 22px;
  font-weight: 400;
}

.c0[data-size='sm'] {
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
}

.c2 {
  margin-right: 8px;
  margin-left: 8px;
  width: 6px;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='10' viewBox='0 0 6 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1 1L5 5L1 9' stroke='%23595959' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
  background-repeat: no-repeat;
  background-size: contain;
  background-position: center;
}

.c2[data-size='lg'] {
  width: 6px;
  margin-right: 8px;
  margin-left: 8px;
}

.c2[data-size='md'] {
  width: 5px;
  margin-right: 7px;
  margin-left: 7px;
}

.c2[data-size='sm'] {
  width: 4px;
  margin-right: 6px;
  margin-left: 6px;
}

.c1 {
  max-width: 12em;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c1 > a {
  color: #262626;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1[data-last='true'] > a {
  color: #737373;
}

.c1 > a:hover,
.c1 > a:focus {
  color: #2c74df;
  outline: none;
}

.c1 > a:hover .arrowLeft,
.c1 > a:focus .arrowLeft,
.c1 > a:hover .arrowRight,
.c1 > a:focus .arrowRight {
  stroke: #2b62cd;
}

.c1 > a:active {
  color: #2b62cd;
}

.c1[data-disabled] > a {
  cursor: not-allowed;
  color: #bfbfbf;
}

<div
  class="c0"
  data-size="md"
>
  <div
    class="c1"
    data-last="false"
    title="Home"
  >
    <a
      href="/"
    >
      Home
    </a>
  </div>
  <div
    class="c2"
    data-size="md"
  />
  <div
    class="c1"
    data-last="false"
    title="Category 1"
  >
    <a
      href="/category-1"
    >
      Category 1
    </a>
  </div>
  <div
    class="c2"
    data-size="md"
  />
  <div
    class="c1"
    data-last="false"
    title="Sub Category 1 with a very long name that will not show completely"
  >
    <a
      href="/category-1/sub-category-1"
    >
      Sub Category 1 with a very long name that will not show completely
    </a>
  </div>
  <div
    class="c2"
    data-size="md"
  />
  <div
    class="c1"
    data-last="true"
    title="Sub Category 2"
  >
    <a
      href="/category-1/sub-category-1/sub-category-2"
    >
      Sub Category 2
    </a>
  </div>
</div>
`;
